From f6d494eb738bb10e779b165496ffbca0c8c86018 Mon Sep 17 00:00:00 2001
From: Nick Desaulniers <ndesaulniers@google.com>
Date: Thu, 13 Jan 2022 12:58:46 -0800
Subject: [PATCH 21/22] ANDROID: arch/Kconfig: fix up LTO LLVM_IAS depdency

Due to LTO being backported to android's 5.10 branches first, the
backport of ba64beb17493a4bfec563100c86a462a15926f24 dropped this. Now
that we have backports for checking AS_IS_LLVM, use that so that we
don't accidentally prevent LTO from being selectable.

Bug: 1938905
Signed-off-by: Nick Desaulniers <ndesaulniers@google.com>
Change-Id: I93fe13dba285bbb9fa1ae0e111bd2955c0b87c12
---
 arch/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/Kconfig b/arch/Kconfig
index d82eafe5d306..5a01c24b7ceb 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -642,7 +642,7 @@ config ARCH_SUPPORTS_LTO_CLANG_THIN
 config HAS_LTO_CLANG
 	def_bool y
 	# Clang >= 11: https://github.com/ClangBuiltLinux/linux/issues/510
-	depends on CC_IS_CLANG && CLANG_VERSION >= 110000 && LD_IS_LLD
+	depends on CC_IS_CLANG && CLANG_VERSION >= 110000 && LD_IS_LLD && AS_IS_LLVM
 	depends on $(success,$(NM) --help | head -n 1 | grep -qi llvm)
 	depends on $(success,$(AR) --help | head -n 1 | grep -qi llvm)
 	depends on ARCH_SUPPORTS_LTO_CLANG
-- 
2.35.1

